### ISSUE TITLE
New phase is added in between instead of at last

### ISSUE THREAD

**Issue by vikasrohit** [appirio-tech/connect-app#2427]
### Expected behavior
As per current requirements it should always add the new phase in the last of project plan.

### Actual behavior
However for projects where there are certain phases which are moved fast or slow than their expected schedule (i.e. when there is a difference in `actualStartDate` and `startDate` of milestones), it tends to add the new phase in between the list of phases.

### Steps to reproduce the problem
- Exact steps are not identified yet

### Screenshot/screencast
Attach or link a resource.

--

#### Environment
- OS: Any
- Browser (w/version): Any
- User role (client, copilot or manager): Copilot or Manager
- Account used: N/A

Aha! Link: https://topcoder.aha.io/features/TCCONNECT-438

**Comment by vikasrohit** [comment]
Moving to next release as it is not reproducible right now.

**Comment by maxceem** [comment]
I've tried to reproduce it several ways playing with activating phases, changing start date and extending, but couldn't.

I've noticed a few things though:

1. When we activate the added phase or existent phase, it goes up to the activated phases. Though not always:

      ![image](https://user-images.githubusercontent.com/146016/45010654-34cccd00-b041-11e8-879f-cb6a34fe72fa.png)

      see http://local.topcoder-dev.com:3000/projects/7262

2. If we extend some phase so it now has the latest end date, this latest end date is ignored by the newly added phase. Logged a separate issue https://github.com/appirio-tech/connect-app/issues/2467. Note. I still cannot reproduce the current issue using the logged issue even it feels like very close to current one.
